cmd/contentpack/main.go

```
@@ -0,0 +1,224 @@
+			}
+
+			metadata.Dependencies = append(metadata.Dependencies,
+				DependencyInformation{contentPackName, contentPackName, hashStr})
+		}
+	}
+
+	metadataBytes, err := yaml.Marshal(&metadata)
+	if err != nil {
+		log.Fatalf("marhsaling metadata.yaml: %s", err)
+	}
+
+	metadataPath := filepath.Join(*outPath, "metadata.yaml")
+	if err := os.WriteFile(metadataPath, metadataBytes, os.ModePerm); err != nil {
+		log.Fatalf("write metadata file for http content pack: %s", err)
+	}
+
+	if err := uploadFolderToS3(*outPath, *awsAccessKeyID, *awsSecretAccessKey, *awsRegion, *awsS3Bucket); err != nil {
+		log.Fatalf("uploading conent packs to %s: %s", *awsS3Bucket, err)
+	}
+}
+
+type DependencyInformation struct {
+	Name      string `yaml:"name"`
+	SourceURI string `yaml:"source_uri"`
+	Hash      string `yaml:"hash"`
+}
+
+type Metadata struct {
+	Name         string                  `yaml:"name"`
+	Dependencies []DependencyInformation `yaml:"dependencies"`
+}
+
+func uploadFolderToS3(folderPath string, awsAccessKeyID string, awsSecretAccessKey string, awsRegion string,
+	awsS3Bucket string,
+) error {
+	creds := credentials.NewStaticCredentials(awsAccessKeyID, awsSecretAccessKey, "")
+	config := aws.Config{Region: aws.String(awsRegion), Credentials: creds}
+
+	awsSession, err := session.NewSession(&config)
+	if err != nil {
+		log.Fatalf("initialize new aws awsSession: %s", err)
+	}
+
+	s3Client := s3.New(awsSession)
+
+	err = filepath.Walk(folderPath, func(path string, info fs.FileInfo, err error) error {
+		if info.IsDir() {
+			return nil
+		}
+		if err := uploadToS3(s3Client, awsS3Bucket, path); err != nil {
+			return errors.Wrapf(err, "uploading content pack %s to S3", path)
+		}
+
+		return nil
+	})
+
+	return err
+}
+
+func calcSha2(inputPath string) (string, error) {
+	f, err := os.Open(inputPath)
+	if err != nil {
+		return "", errors.Wrapf(err, "open zip %s content pack", inputPath)
+	}
+	defer f.Close()
```

Better not to ignore errors, we can create a helper function like this and reuse it everywhere in this file
```
func close(elem io.Closer) {
	err := elem.Close()
	if err != nil {
		log.Warnf("failed to close file: %v", err)
	}
}
```

---

nice, will call it closeD (D for deferred). log.Warnf introduced in later golang version AFAIK so I'll need to update our min. supported golang for this in go.mod

---

Okay cool
And you can just change the standard log library "log" to `log "github.com/sirupsen/logrus"
`
We use it almost everywhere and it supports a few more things